### PR TITLE
Implement srread function

### PR DESCRIPTION
**Pull request by CPBridge**
Adding a convenience function `highdicom.sr.srread`, equivalent to `highdicom.seg.segread`, capable of reading in any SR document supported by highdicom.

### CODE REVIEW

File: src/highdicom/sr/sop.py [hunk 1]
```diff
@@ -770,3 +783,48 @@ def from_dataset(
...
+def srread(
+    fp: Union[str, bytes, PathLike, BinaryIO],
+) -> Union[EnhancedSR, ComprehensiveSR, Comprehensive3DSR]:
+    """Read a structured report document in DICOM File Format.
+
+    The object is returned as an instance of the highdicom class corresponding
+    to the dataset's IOD. Currently supported IODs are:
+    - EnhancedSRStorage
+    - ComprehensiveSRStorage
+    - Comprehensive3DSRStorage
```
Comment (hackermd) [inline src/highdicom/sr/sop.py:797]: Are you sure the list will render correctly?

File: src/highdicom/sr/sop.py [hunk 2]
```diff
@@ -770,3 +783,48 @@ def from_dataset(
...
+
+    Raises
+    ------
+    RuntimeError:
+        If the DICOM file has an IOD not supported by highdicom.
+
+    Returns
+    -------
+    Union[highdicom.sr.EnhancedSR, highdicom.sr.ComprehensiveSR, highdicom.sr.Comprehensive3DSR]
+        Segmentation image read from the file.
```
Comment (hackermd) [inline src/highdicom/sr/sop.py:813]:
```suggestion
        Structured Report document read from the file.
```

File: src/highdicom/sr/sop.py [hunk 3]
```diff
@@ -770,3 +783,48 @@ def from_dataset(
         sop_instance = super().from_dataset(dataset, copy=copy)
         sop_instance.__class__ = Comprehensive3DSR
         return cast(Comprehensive3DSR, sop_instance)
+
+
+def srread(
+    fp: Union[str, bytes, PathLike, BinaryIO],
+) -> Union[EnhancedSR, ComprehensiveSR, Comprehensive3DSR]:
+    """Read a structured report document in DICOM File Format.
```
Comment (hackermd) [inline src/highdicom/sr/sop.py:791]:
```suggestion
    """Read a Structured Report (SR) document in DICOM File format.
```

**Review by hackermd — CHANGES_REQUESTED** [review]

File: src/highdicom/sr/sop.py (hunk 1 above)
Comment (CPBridge) [reply src/highdicom/sr/sop.py]: Good point, resolved in c436b18

File: src/highdicom/sr/sop.py [hunk 4]
```diff
@@ -770,3 +783,49 @@ def from_dataset(
...
+    fp: Union[str, bytes, PathLike, BinaryIO],
+) -> Union[EnhancedSR, ComprehensiveSR, Comprehensive3DSR]:
+    """Read a Structured Report (SR) document in DICOM File format.
+
+    The object is returned as an instance of the highdicom class corresponding
+    to the dataset's IOD. Currently supported IODs are:
+
+    * EnhancedSRStorage via class :class:`EnhancedSR`
+    * ComprehensiveSRStorage via class :class:`ComprehensiveSR`
+    * Comprehensive3DSRStorage via class :class:`Comprehensive3DSR`
```
Comment (hackermd) [inline src/highdicom/sr/sop.py:798]: The names of the IODs are as follows:

```suggestion
    * Enhanced SR via class :class:`EnhancedSR`
    * Comprehensive SR via class :class:`ComprehensiveSR`
    * Comprehensive 3D SR via class :class:`Comprehensive3DSR`
```

File: src/highdicom/sr/sop.py [hunk 5]
```diff
@@ -770,3 +783,49 @@ def from_dataset(
...
+        ComprehensiveSRStorage: ComprehensiveSR,
+        Comprehensive3DSRStorage: Comprehensive3DSR,
+    }
+    dcm = dcmread(fp)
+
+    sopuid = dcm.SOPClassUID
+    if sopuid in class_map:
+        return class_map[sopuid].from_dataset(dcm, copy=False)
+    else:
+        iod_name = UID_dictionary[sopuid][0]
```
Comment (hackermd) [inline src/highdicom/sr/sop.py:828]:
```suggestion
    sop_class_uid = dcm.SOPClassUID
    if sop_class_uid in class_map:
        return class_map[sop_class_uid].from_dataset(dcm, copy=False)
    else:
        iod_name = UID_dictionary[sop_class_uid][0]
```

File: src/highdicom/sr/sop.py [hunk 6]
```diff
@@ -770,3 +783,49 @@ def from_dataset(
...
+        raise RuntimeError(
+            f'SOP Class UID {dcm.SOPClassUID} "{iod_name}" is not supported.'
```
Comment (hackermd) [inline src/highdicom/sr/sop.py:830]:
```suggestion
            f'SOP Class UID {sop_class_uid} "{iod_name}" is not supported.'
```

File: tests/test_sr.py [hunk 7]
```diff
@@ -3877,6 +3878,27 @@ def test_evidence_multiple_studies(self):
...
+            manufacturer=self._manufacturer,
+            performed_procedure_codes=self._performed_procedures,
+        )
+
+        with BytesIO() as buf:
+            report.save_as(buf)
+            buf.seek(0)
+            sr_read = srread(buf)
+
+        assert isinstance(sr_read, ComprehensiveSR)
```
Comment (hackermd) [inline tests/test_sr.py:3900]:
```suggestion
            sr_instance = srread(buf)

        assert isinstance(sr_instance, ComprehensiveSR)
```

File: tests/test_sr.py [hunk 8]
```diff
@@ -4925,6 +4947,14 @@ def test_get_groups_with_ref_uid_and_graphic_type_3d(self):
                 referenced_sop_instance_uid=UID()
             )
 
+    def test_srread(self):
+        with BytesIO() as buf:
+            self._sr.save_as(buf)
+            buf.seek(0)
+            sr_read = srread(buf)
+
+        assert isinstance(sr_read, Comprehensive3DSR)
```
Comment (hackermd) [inline tests/test_sr.py:4956]:
```suggestion
            sr_instance = srread(buf)

        assert isinstance(sr_instance, Comprehensive3DSR)
```

**Review by hackermd — CHANGES_REQUESTED** [review]
LGTM. Just a few suggestions regarding variable names to improve clarity.

**Review by hackermd — APPROVED** [review]

**Pull request merged**